test/periphery/CardPaymentProcessorUpgradeable.test.ts

```diff
@@ -0,0 +1,1421 @@
+    const tokenBalance: number = payments.map(function (payment: TestPayment): number {
+      return payment.status == PaymentStatus.Uncleared || payment.status == PaymentStatus.Cleared ? payment.amount : 0
+    }).reduce((sum: number, current: number) => sum + current);
+    const totalClearedBalance: number = payments.map(function (payment: TestPayment): number {
+      return payment.status == PaymentStatus.Cleared ? payment.amount : 0
+    }).reduce((sum: number, current: number) => sum + current);
+    const totalUnclearedBalance: number = payments.map(function (payment: TestPayment): number {
+      return payment.status == PaymentStatus.Uncleared ? payment.amount : 0
+    }).reduce((sum: number, current: number) => sum + current);
+    const clearedBalancesPerAccount: Map<string, number> = defineBalancesPerAccount(
+      payments,
+      PaymentStatus.Cleared
+    );
+    const unclearedBalancesPerAccount: Map<string, number> = defineBalancesPerAccount(
+      payments,
+      PaymentStatus.Uncleared
+    );
+
+    return {
+      tokenBalance,
+      totalUnclearedBalance,
+      totalClearedBalance,
+      clearedBalancesPerAccount,
+      unclearedBalancesPerAccount
+    };
+  }
+
+  async function checkPaymentStructuresOnBlockchain(payments: TestPayment[]) {
+    for (let i = 0; i < payments.length; ++i) {
+      const expectedPayment: TestPayment = payments[i];
+      const actualPayment = await cardPaymentProcessor.paymentFor(
+        createBytesString(expectedPayment.authorizationId, BYTES16_LENGTH)
+      );
+      checkEquality(actualPayment, expectedPayment, i);
+      if (!!expectedPayment.parentTxHash) {
+        expect(await cardPaymentProcessor.isPaymentReversed(expectedPayment.parentTxHash)).to.equal(
+          expectedPayment.status == PaymentStatus.Reversed
+        );
+        expect(await cardPaymentProcessor.isPaymentRevoked(expectedPayment.parentTxHash)).to.equal(
+          expectedPayment.status == PaymentStatus.Revoked
+        )
+      }
+    }
+  }
+
+  async function checkBalancesOnBlockchain(
+    expectedBalancesPerAccount: Map<string, number>,
+    isClearedBalance: boolean
+  ) {
+    for (const account of expectedBalancesPerAccount.keys()) {
+      const expectedBalance = expectedBalancesPerAccount.get(account);
+      if (!expectedBalance) {
+        continue;
+      }
+      if (isClearedBalance) {
+        expect(await cardPaymentProcessor.clearedBalanceOf(account)).to.equal(
+          expectedBalance,
+          `The cleared balance for account ${account} is wrong`
+        )
+      } else {
+        expect(await cardPaymentProcessor.unclearedBalanceOf(account)).to.equal(
+          expectedBalance,
+          `The uncleared balance for account ${account} is wrong`
+        )
+      }
+    }
+  }
+
+  async function checkCardPaymentProcessorState(payments: TestPayment[]) {
+    const expectedState: CardPaymentProcessorState = defineExpectedCardPaymentProcessorState(payments);
```

Comment:
Codacy has a fix for the issue: [Insert `⏎·····`](https://app.codacy.com/gh/cloudwalk/brlc-token/pullRequest?prid=9781596)
```suggestion
    const expectedState: CardPaymentProcessorState =
```